### ISSUE TITLE
Malli generates incorrect clj-kondo type spec for tuples

### ISSUE THREAD

**Issue by divs1210** [metosin/malli#826]
```clojure
(def location
  [:tuple {:title "location"}
   number? number?])

(m/=> location->str [:=> [:cat location]
                         string?])
(defn location->str [[x y]]
  (str "(" x ", " y ")"))

((requiring-resolve 'malli.clj-kondo/emit!))

(location->str [0 0]) ;; Expected: number, received: vector  
```

**Comment by opqdonut** [comment]
This should get fixed by #851

**Comment by divs1210** [comment]
Just checked again, this works now! 🎉
